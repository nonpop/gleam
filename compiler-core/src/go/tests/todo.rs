use crate::assert_go;

#[test]
fn without_message() {
    assert_go!(
        r#"
fn go() {
    todo
}
"#,
    );
}

#[test]
fn with_message() {
    assert_go!(
        r#"
fn go() {
  todo as "I should do this"
}
"#,
    );
}

#[test]
fn with_message_expr() {
    assert_go!(
        r#"
fn go() {
  let x = "I should " <> "do this"
  todo as x
}
"#,
    );
}

// https://github.com/gleam-lang/gleam/issues/1238
#[test]
fn as_expression() {
    assert_go!(
        r#"
fn go(f) {
  let boop = todo as "I should do this"
  f(todo as "Boom")
}
"#,
    );
}
