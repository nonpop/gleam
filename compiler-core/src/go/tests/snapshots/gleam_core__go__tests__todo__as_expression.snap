---
source: compiler-core/src/go/tests/todo.rs
expression: "\nfn go(f) {\n  let boop = todo as \"I should do this\"\n  f(todo as \"Boom\")\n}\n"
snapshot_kind: text
---
----- SOURCE CODE

fn go(f) {
  let boop = todo as "I should do this"
  f(todo as "Boom")
}


----- COMPILED GO
package mod_P

import (
  gleam_P "example.com/todo/gleam"
)

const _ = gleam_P.Use_Import
const Use_Import byte = 0

func go π[M gleam_P.Type[M], N gleam_P.Type[N]](f gleam_P.Func1_t[N, M]) M {
  var boop gleam_P.Type = (func() gleam_P.Type {
    panic(gleam_P.MakeError(
      "todo",
      "my/mod",
      3,
      "go",
      "I should do this",
      map[string]any{},
    ))
  })()
  _ = boop
  return f(
    (func() N {
      panic(gleam_P.MakeError(
        "todo",
        "my/mod",
        4,
        "go",
        "Boom",
        map[string]any{},
      ))
    })(),
  )
}
